From 702963cff52863b1870c06c61a5427339a9852ad Mon Sep 17 00:00:00 2001
From: merothh <merothh@gmail.com>
Date: Sun, 26 Jul 2020 05:30:36 +0530
Subject: [PATCH] [DNM]: phoenix: use walleye fingerprint

---
 init/init_phoenix.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init/init_phoenix.cpp b/init/init_phoenix.cpp
index 72c592e..8ad359e 100644
--- a/init/init_phoenix.cpp
+++ b/init/init_phoenix.cpp
@@ -68,9 +68,9 @@ constexpr const char *BUILD_DESCRIPTION[] = {
 };
 
 constexpr const char *BUILD_FINGERPRINT[] = {
-    "Redmi/phoenix/phoenix:10/QKQ1.190825.002/V12.0.3.0.QGHCNXM:user/"
+    "google/walleye/walleye:8.1.0/OPM1.171019.011/4448085:user/"
     "release-keys",
-    "POCO/phoenixin/phoenixin:10/QKQ1.190825.002/V11.0.11.0.QGHINXM:user/"
+    "google/walleye/walleye:8.1.0/OPM1.171019.011/4448085:user/"
     "release-keys",
 };
 
-- 
2.25.1

